operators/src/upgrade/k8s/crd/mod.rs

```diff
@@ -0,0 +1,14 @@
+use lazy_static::lazy_static;
+pub use semver::Version;
+pub mod v0;
+
+lazy_static! {
+    // Regex gathered from semver.org as the recommended semver validation regex.
+    static ref SEMVER_RE: regex::Regex = regex::Regex::new(
```

Comment:
I'm confused, if we're using the semver crate already, why are we using our own regex?

Reply:
We would have to implement JsonSchema, Serialize and Deserialize for Version.

Reply:
To enable `Serialize/Deserialize` you have to enable the `serde` feature in the `semvercrate`. Why do you need JsonSchema? Change ` current_version: Option<String>` to ` current_version: semver::Version,` and it becomes required?

Reply:
JsonSchema is required for Spec of CRD. 
And to use semver::Version we will have to implement JsonSchema for semver::Version.

Reply:
If that's too much to get it working then I'm fine with the regex for now.